REST-API-ADVANCED/advanced/src/main/java/com/epam/esm/certificate/service/CertificateServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package com.epam.esm.certificate.service;
+
+import com.epam.esm.certificate.entity.Certificate;
+import com.epam.esm.certificate.exception.CertificateInvalidRequestException;
+import com.epam.esm.certificate.exception.CertificateNotFoundException;
+import com.epam.esm.certificate.repo.CertificateRepository;
+import com.epam.esm.errorhandle.constants.ErrorConstants;
+import com.epam.esm.errorhandle.validation.Validate;
+import com.epam.esm.tag.entity.Tag;
+import com.epam.esm.tag.service.TagService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CertificateServiceImpl implements CertificateService {
+
+    private final CertificateRepository certificateRepository;
+    private final TagService tagService;
+    private final Validate validate;
+
+    @Autowired
+    public CertificateServiceImpl(CertificateRepository certificateRepository, TagService tagService, Validate validate) {
+        this.certificateRepository = certificateRepository;
+        this.tagService = tagService;
+        this.validate = validate;
+    }
+
+    public List<Certificate> getAllCertificate(Pageable paginationCriteria) {
+        return certificateRepository.findAll(
+                validate.validNonErroneousPageableRequest(
```

Comment:
`validNonErroneousPageableRequest`? Is there a `validErroneousPageableRequest` or `invalidNonErroneousPageableRequest`? What's the difference?

Reply:
i agree that sounds confusing. Renamed validNonErroneousPageableRequest to validPageableRequest.